src/semantic_world/connection_factories.py

```diff
@@ -0,0 +1,211 @@
+    """
+
+    name: PrefixedName
+    parent_name: PrefixedName
+    child_name: PrefixedName
+
+    @classmethod
+    def from_connection(cls, connection: Connection) -> Self:
+        for factory in recursive_subclasses(cls):
+            if factory.original_class() == connection.__class__:
+                return factory._from_connection(connection)
+        raise ValueError(f"Unknown connection type: {connection.name}")
+
+    @classmethod
+    @abstractmethod
+    def _from_connection(cls, connection: Connection) -> Self:
+        """
+        Create a connection factory from a connection.
+
+        :param connection: The connection to create the factory from.
+        :return: The created connection factory.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def create(self, world: World) -> T:
+        """
+        Create the connection in a given world.
+
+        :param world: The world in which to create the connection.
+        :return: The created connection.
+        """
+        raise NotImplementedError
+
+    def to_json(self) -> Dict[str, Any]:
+        return {
+            **super().to_json(),
+            "name": self.name.to_json(),
+            "parent_name": self.parent_name.to_json(),
+            "child_name": self.child_name.to_json(),
+        }
+
+@dataclass
+class FixedConnectionFactory(ConnectionFactory[FixedConnection]):
+
+    @classmethod
+    def _from_connection(cls, connection: Connection) -> Self:
+        return cls(name=connection.name,
+                   parent_name=connection.parent.name,
+                   child_name=connection.child.name,
+                   )
+
+    def create(self, world: World) -> Connection:
+        parent = world.get_kinematic_structure_entity_by_name(self.parent_name)
+        child = world.get_kinematic_structure_entity_by_name(self.child_name)
+        return self.original_class()(parent=parent, child=child, name=self.name)
+
+    @classmethod
+    def _from_json(cls, data: Dict[str, Any]) -> Self:
+        return cls(name=PrefixedName.from_json(data["name"]),
+                   parent_name=PrefixedName.from_json(data["parent_name"]),
+                   child_name=PrefixedName.from_json(data["child_name"]),
+                   )
+
+@dataclass
+class ActiveConnection1DOFFactory(ConnectionFactory[T]):
+    axis: cas.Vector3
+    multiplier: float
+    offset: float
+    dof: DegreeOfFreedom
+
+    @classmethod
+    def _from_connection(cls, connection: PrismaticConnection) -> Self:
+        return cls(name=connection.name,
+                   parent_name=connection.parent.name,
+                   child_name=connection.child.name,
+                   axis=connection.axis,
+                   multiplier=connection.multiplier,
+                   offset=connection.offset,
+                   dof=connection.dof,
+                   )
+
+    def create(self, world: World) -> Connection:
```

Comment:
You also need to create a new dof since that also contains back references to the world

Reply:
Yes but connections can share DOF which would be lost in this case